src/utilities/network/lookupApi.js

```diff
@@ -0,0 +1,110 @@
+import { UIDatabase } from '../../database/index';
+
+const { SYNC_URL } = SETTINGS_KEYS;
+
+const RESOURCES = {
+  PATIENT: '/api/v4/patient',
+  PRESCRIBER: '/api/v4/prescriber',
+};
+
+const SEPARATORS = {
+  QUERY_STRING: '?',
+  QUERY_PARAMETERS: '&',
+  POLICY_NUMBER: '-',
+};
+
+const getQueryString = params => {
+  const {
+    firstName = '',
+    lastName = '',
+    registrationCode = '',
+    dateOfBirth = '',
+    policyNumberPerson = '',
+    policyNumberFamily = '',
+  } = params ?? {};
+  const policyNumberSeparator =
+    policyNumberPerson && policyNumberFamily ? SEPARATORS.POLICY_NUMBER : '';
+  const policyNumber = policyNumberPerson + policyNumberSeparator + policyNumberFamily;
+  const queryParams = [
+    { first_name: firstName },
+    { last_name: lastName },
+    { code: registrationCode },
+    { dob: dateOfBirth },
+    { policy_number: policyNumber },
+  ];
+  return queryParams.reduce((queryString, param) => {
+    const [[key, value]] = Object.entries(param);
+    if (!value) return queryString;
+    const paramString = `${key}=${value}`;
+    const paramSeparator = queryString.length > 0 ? SEPARATORS.PARAMETERS : SEPARATORS.QUERY_STRING;
+    return queryString + paramSeparator + paramString;
+  }, '');
+};
+
+const getPatientRequestUrl = params => {
+  const baseUrl = UIDatabase.getSetting(SYNC_URL);
+  const endpoint = RESOURCES.PATIENT;
+  const queryString = getQueryString(params);
+  return baseUrl + endpoint + queryString;
+};
+
+const getPrescriberRequestUrl = params => {
+  const baseUrl = UIDatabase.getSetting(SYNC_URL);
+  const endpoint = RESOURCES.PRESCRIBER;
+  const queryString = getQueryString(params);
+  return baseUrl + endpoint + queryString;
+};
+
+const processPatientResponse = async response => {
+  const responseJson = await response.json();
+  const { error } = responseJson;
+  if (error) throw new Error(error);
+  const patientData = responseJson.map(patient => ({
+    firstName: patient.first,
+    lastName: patient.last,
+    dateOfBirth: new Date(patient.date_of_birth),
+  }));
+  return patientData;
+};
+
+const processPrescriberResponse = async response => {
+  const responseJson = await response.json();
+  const { error } = responseJson;
+  if (error) throw new Error(error);
+  const prescriberData = responseJson.map(prescriber => ({
+    firstName: prescriber.first,
+    lastName: prescriber.last,
+    registrationCode: prescriber.code,
+  }));
+  return prescriberData;
+};
+
+export const queryPatientApi = async params => {
+  const requestUrl = getPatientRequestUrl(params);
+  try {
+    const response = await fetch(requestUrl);
+    const patientData = await processPatientResponse(response);
+    return patientData;
+  } catch (error) {
+    // TODO: add bugsnag.
+    return [];
+  }
+};
+
+export const queryPrescriberApi = async params => {
+  const requestUrl = getPrescriberRequestUrl(params);
+  try {
+    const response = await fetch(requestUrl);
```

Comment:
I thought `fetch` didn't throw on error responses - so you'd need to check if the response status was in `[200, 300)`?

Also this must need authorization header?

Reply:
Yeah, still very basic. Definitely needs more response handling logic. Same goes for authorisation header (but I think this requires some more work on desktop). I think both cases of for new issues (going to do an issue blitz asap once I've done a bit more testing).

Reply:
Desktop PR made for authentication logic (https://github.com/sussol/msupply/pull/5905) and epic #2446 updated with issue.